src/theory/arith/arith_rewriter.cpp

```diff
@@ -968,7 +968,8 @@ RewriteResponse ArithRewriter::postRewriteIntsLog2(TNode t)
     Assert(t[0].getType().isInteger());
     Integer i = t[0].getConst<Rational>().getNumerator();
     size_t const length = i.length();
-    return RewriteResponse(REWRITE_DONE, rewriter::mkConst(Integer(length)));
+    return RewriteResponse(REWRITE_DONE,
+                           rewriter::mkConst(Integer(length - 1)));
```

Comment:
Interesting bug. Maybe add an assertion that `i` is positive? (unless it's defined for non-positive numbers, in which case I would prefer the result to be 0)

Reply:
I've update to make log of negative equal to 0.

Reply:
What about 0?

Reply:
log_2 0 was zero already.